azurerm/resource_arm_app_service_certificate_order.go

```diff
@@ -0,0 +1,488 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppServiceCertificateOrder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceCertificateOrderCreateUpdate,
+		Read:   resourceArmAppServiceCertificateOrderRead,
+		Update: resourceArmAppServiceCertificateOrderCreateUpdate,
+		Delete: resourceArmAppServiceCertificateOrderDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"auto_renew": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Optional: true,
+			},
+
+			"certificates": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"certificate_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"key_vault_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"key_vault_secret_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"provisioning_state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"csr": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"distinguished_name"},
+			},
+
+			"distinguished_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"csr"},
+			},
+
+			"key_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2048,
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+
+			"product_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "standard",
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc: validation.StringInSlice([]string{
+					"standard",
+					"wildcard",
+				}, true),
+			},
+
+			"validity_in_years": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 3),
+			},
+
+			"domain_verification_token": {
```

Comment:
Terraform doesn't need to know about every attribute an API returns but instead just attributes that directly affect the resource or could be used in another resource. With that in mind, I don't think any of these `Computed` attributes need to be specified in the schema. I'm open to further discussion though if you believe otherwise.

Reply:
Remove the computed attributes seem useless.